src/server/routes/admin/nonces.ts

```diff
@@ -0,0 +1,192 @@
+import { getChain } from "../../../utils/chain";
+import { normalizeAddress } from "../../../utils/primitiveTypes";
+import { redis } from "../../../utils/redis/redis";
+import { thirdwebClient } from "../../../utils/sdk";
+import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+import { walletWithAddressParamSchema } from "../../schemas/wallet";
+
+export const responseBodySchema = Type.Object({
+  result: Type.Array(
+    Type.Object({
+      walletAddress: Type.String({
+        description: "Backend Wallet Address",
+        examples: ["0xcedf3b4d8f7f1f7e0f7f0f7f0f7f0f7f0f7f0f7f"],
+      }),
+      chainId: Type.Number({
+        description: "Chain ID",
+        examples: [80002],
+      }),
+      onchainNonce: Type.Integer({
+        description: "Last mined nonce",
+        examples: [0],
+      }),
+      lastUsedNonce: Type.Integer({
+        description: "Last incremented nonce sent to the RPC",
+        examples: [0],
+      }),
+      sentNonces: Type.Array(Type.Integer(), {
+        description:
+          "Nonces that were successfully sent to the RPC but not mined yet (in descending order)",
+        examples: [[2, 1, 0]],
+      }),
+      recycledNonces: Type.Array(Type.Integer(), {
+        examples: [[3, 2, 1]],
+        description:
+          "Nonces that were acquired but failed to be sent to the blockchain, waiting to be recycled or cancelled (in descending order)",
+      }),
+    }),
+  ),
+});
+
+responseBodySchema.example = {
+  result: [
+    {
+      walletAddress: "0xcedf3b4d8f7f1f7e0f7f0f7f0f7f0f7f0f7f0f7f",
+      onchainNonce: 2,
+      lastUsedNonce: 8,
+      recycledNonces: [6, 7],
+      chainId: 80002,
+    },
+  ],
+};
+
+const walletWithAddressQuerySchema = Type.Partial(walletWithAddressParamSchema);
+
+export async function getNonceDetailsRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Querystring: Static<typeof walletWithAddressQuerySchema>;
+    Reply: Static<typeof responseBodySchema>;
+  }>({
+    method: "GET",
+    url: "/admin/nonces",
+    schema: {
+      summary: "Get nonce status details for wallets",
+      description:
+        "Admin route to get nonce status details for all wallets filtered by address and chain ",
+      tags: ["Admin"],
+      operationId: "nonceDetails",
+      querystring: walletWithAddressQuerySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseBodySchema,
+      },
+      hide: true,
+    },
+    handler: async (request, reply) => {
+      const { walletAddress, chain } = request.query;
+      const result = await getNonceDetails({ walletAddress, chainId: chain });
+
+      reply.status(StatusCodes.OK).send({
+        result,
+      });
+    },
+  });
+}
+
+export const getNonceDetails = async ({
+  walletAddress,
+  chainId,
+}: {
+  walletAddress?: string;
+  chainId?: string;
+} = {}) => {
+  const lastUsedNonceKeys = await getLastUsedNonceKeys(walletAddress, chainId);
```

Comment:
Can we use `getUsedBackendWallets()` here? I think `getLastUsedNonceKeys()` doesn't need to exist anymore